### ISSUE TITLE
Geo-Preprocessing Module

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/oceanbench#3]
**TLDR**

This will introduce some geo-preprocessing functionality. It will slice, dice and process the data to a more application specific state.

---
**Example Steps**

1. **Change Domain** (Space): spherical, Cartesian, Local Tangent Plane
2. **Slice Domain**: `slice(2011,2021)`
3. **Discretise** (Spatial): `coarsen`
4. **Discretize** (Temporal): `rolling`
5. Gridify AlongTrack Observations
6. Alongtrack-ify Grid Observations
7. Normalisation, e.g. remove climatology
8. Filtering, e.g. `gaussfilter(ds)`
9. Noising, e.g. `noise(ds, "gauss")`
10. Calculate higher level variables, e.g. `u,v,dlon,dlat,dx,dy,dt`
11. Select appropriate variables, e.g. `ssh,sst,sss,u,v,x,y,time`

---
**Suggested Functionality**

- [ ] Spatial Domain Change - Spherical Cartesian, Local Tangent Plane!
- [ ] Temporal Domain Change
- [ ] Discretization - coarsening, alongtrack
- [x] #13
- [ ] Alongtrackify
- [ ] Normalizing
- [ ] Filtering
- [ ] Noising
- [ ] Derived Variables
- [ ] u,v,ssc
- [ ] ssh,potential vortcity
